### PR TITLE
Adding omitted notes, mostly on syntax

### PR DESCRIPTION
**Pull request by catb0t**
This has more commits yet to come, in Pair.md for example, but i'll open it before i go to bed.

the `Number!PREC` syntax ought to get documentation too, but i'll add a bunch of operator docs at once and not in this pull / series.

### CODE REVIEW

File: syntax_and_semantics/variables.md [hunk 1]
```diff
@@ -126,14 +126,30 @@ say x              # prints: 15
 
 ## Special variables
 
-Currently, there are only two real predefined variables: `ARGV` and `ENV`.
+Currently, there are only four real predefined variables:
+
+* `Array ARGV`: The program's command-line arguments, that were not given to Sidef
+* `Hash ENV`: Writable copy of environment variables and their values when the program was started.
+* `FileHandle ARGF`: Used to read lines from argument-files or from `STDIN` when no argument has been specified
+* `FileHandle DATA`:  Contains The data stored after the __END__ or __DATA__ tokens
```
Comment (catb0t) [inline syntax_and_semantics/variables.md]: oops, Contains The

File: syntax_and_semantics/literals/nil_and_null.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+Instead of an error or exception, an *undefined* Hash key has the value `nil`.
+
+```ruby
+Hash(){:key} # -> nil
+```
+
+## Null
+
+The *Null* singleton type represents a defined value, with no specific meaning other than its existence.
```
Comment (catb0t) [inline syntax_and_semantics/literals/nil_and_null.md]: add phrasing "context-dependent"

File: syntax_and_semantics/literals/pair.md [hunk 3]
```diff
@@ -1,8 +1,116 @@
...
+"paired":"strings"        # -> Pair("paired", "strings")
+Pair("paired", "strings") # -> ==/==
+
+"a":nil                   # -> Pair("a", nil)
+nil:"b"                   # -> error, methods cannot be called on undefined values
+Pair(nil, "b")            # -> Pair(nil, "b")
+```
+Note that the Pair literal syntax does not round-trip, i.e `Pair.to_s` will never give the `:` representation.
+
+When using the Pair literal syntax with variable names, this overlaps with the syntax for Complex numbers and NamedParams.
```
Comment (catb0t) [inline syntax_and_semantics/literals/pair.md:21]: reword to clarify that `Number.:` is a method whereas `name:` is NamedParam operator syntax

File: syntax_and_semantics/literals/pair.md [hunk 4]
```diff
@@ -1,8 +1,116 @@
...
+say (a : b)             # ==/==
+say Pair(a, b)          # ==/==
+
+say c.:d                # prints 1+2i
+say c\:d                # ==/==
+say (c : d)             # ==/==
+say Pair(c, d)          # prints Pair(1, 2)
+```
+
+Note this does not override the Complex literal when the left-hand side *is* a number.
```
Comment (catb0t) [inline syntax_and_semantics/literals/pair.md]: ditto

**Comment by catb0t** [comment]
Literals / typenames still missing documentation:

```
Try 
Fork 
Stat 
SocketHandle 
Pipe 
Backtick
Range 
RangeNumber 
RangeString
Regex
Match
```
I probably won't add too any more of these yet

**Comment by trizen** [comment]
Fantastic work! Thank you very much! :+1:

**Pull request merged**